cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueClientFactory.java

```diff
@@ -234,7 +234,22 @@ private HBaseQueueAdmin ensureTableExists(QueueName queueName) throws IOExceptio
     return admin;
   }
 
-  private HTable createHTable(TableId tableId) throws IOException {
+  /**
+   * Helper method to check if table exists.
+   * @param queueName name of the queue to check.
+   * @return true if the table exists.
+   * @throws IOException
+   */
+  boolean tableExists(QueueName queueName) throws IOException {
```

Comment:
Do you need this? Can you just rely on the `createHTable` to throw a `TableNotFoundException` to indicate the queue table doesn't not exists?